src/phone-input/country-select-dropdown.js

```diff
@@ -60,59 +101,80 @@ function CountrySelectDropdown(
     overrides.CountrySelectDropdownDialcodeColumn,
     DefaultDialcodeColumn,
   );
-  const scrollIndex = Math.min(
-    children.findIndex(opt => opt.props.item.id === country.id) + 5,
-    children.length - 1,
+  const [FilterInput, filterInputProps] = getOverrides(
+    overrides.FilterInput,
+    DefaultFilterInput,
   );
   return (
-    <Container
+    <PopoverContentContainer
       ref={forwardedRef}
-      $height={maxDropdownHeight}
-      {...containerProps}
+      {...popoverContentContainerProps}
     >
-      <AutoSizer>
-        {({height, width}) => {
-          return (
-            <List
-              role="listbox"
-              height={height}
-              width={width}
-              rowCount={children.length}
-              rowHeight={42}
-              scrollToIndex={scrollIndex}
-              rowRenderer={({index, key, style}) => {
-                // resetMenu and getItemLabel should not end up on native html elements
-                const {resetMenu, getItemLabel, ...rest} = children[
-                  index
-                ].props;
-                return (
-                  <ListItem
-                    key={key}
-                    style={style}
-                    {...rest}
-                    {...listItemProps}
-                    data-e2e="country-select-list-item"
-                    data-iso={children[index].props.item.id}
-                  >
-                    <FlagColumn {...flagColumnProps}>
-                      <StyledFlag iso={children[index].props.item.id} />
-                    </FlagColumn>
-                    <NameColumn {...nameColumnProps}>
-                      {mapIsoToLabel
-                        ? mapIsoToLabel(props.children[index].props.item.id)
-                        : children[index].props.item.label}
-                    </NameColumn>
-                    <Dialcode {...dialcodeProps}>
-                      {children[index].props.item.dialCode}
-                    </Dialcode>
-                  </ListItem>
-                );
-              }}
-            />
-          );
-        }}
-      </AutoSizer>
-    </Container>
+      {enableFiltering && (
+        <FilterInput
+          onChange={event => {
+            onFilter(event);
+            setScrollIndex(0);
+          }}
+          {...filterInputProps}
+        />
+      )}
+      <Container
+        $height={calculateDropDownHeight(
+          maxDropdownHeight,
+          children.length,
+          LIST_ROW_HEIGHT,
```

Comment:
Not sure about using a constant here for the row height. Is there a case when the height is larger/smaller due to text wrapping or font change?

Reply:
We need this constant here to calculate the correct height of dropdown for example when only 2 items satisfy filter condition. I don't saw these cases but maybe @sandgraham know more?